### PR TITLE
feat(allowance): add option to change lnurlauth

### LINKED ISSUE #148: Display auto login option for publishers
The publisher list and detail page and popup should show of auto login with lnurl-auth is enabled or not.

Similar to updating the budget it should also be possible to enable/disable the auto login from the publishers edit form.

image
image

### PR DESCRIPTION
**Pull request by im-adithya**
### Describe the changes you have made in this PR

Adds option to set lnurlAuth for allowances from Publishers screen.

### Link this PR to an issue

Closes #148

### Type of change (Remove other not matching type)

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes (If any)

https://user-images.githubusercontent.com/64399555/192765505-d4d6c4d0-ca98-4a7e-97c8-10ff59f317d5.mp4

### How has this been tested?

Manually

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Comment by escapedcat** [comment]
> When enabled, will automatically login on a site? If not enabled I will always be asked to confirm login? 

Yes

> The way it works now, I think I'm no asked to confirm login if I logout and login again.

Correct, currently the default is "enabled" after you logged in once.
This PR will change this so a user can chose to logged in automatically or not.

File: src/app/components/AllowanceMenu/index.tsx [hunk 1]
```diff
@@ -138,6 +142,20 @@ function AllowanceMenu({ allowance, onEdit, onDelete }: Props) {
...
               />
             </div>
+            <div className="flex items-center mt-6">
+              <Checkbox
+                id="enable_login"
+                name="enable_login"
+                checked={login}
+                onChange={(e) => setLogin(!login)}
+              />
+              <label
```
Comment (escapedcat) [inline src/app/components/AllowanceMenu/index.tsx:152]: This is the first time we use a checkbox. For everything else we use `Toggle`. Should we use toggle here as well?

Comment (im-adithya) [reply src/app/components/AllowanceMenu/index.tsx:152]: We also use it in Remember and Set Budget right?

Comment (im-adithya) [reply src/app/components/AllowanceMenu/index.tsx:152]: But yeah, I think it's not being highlighted much right now, may be we can use toggle and also add hint below it (like we do in settings) to let people know what it is about

**Comment by bitcoinuser** [comment]
> Correct, currently the default is "enabled" after you logged in once.
This PR will change this so a user can chose to logged in automatically or not.

I really don't see a reason to have this feature. I think more confuse than helps. What do you think @bumi?

**Comment by im-adithya** [comment]
I think this is more about giving control to the user to toggle the lnurlAuth option and would make more sense once the follow-up PR to this comes up which solves #1534 where we show the websites where we used lightning login and now if the user wants to disable it in any website then he can choose to edit the allowance

File: src/app/components/AllowanceMenu/index.tsx [hunk 2]
```diff
@@ -29,6 +30,7 @@ function AllowanceMenu({ allowance, onEdit, onDelete }: Props) {
 
   const [modalIsOpen, setIsOpen] = useState(false);
   const [budget, setBudget] = useState("0");
+  const [login, setLogin] = useState(false);
```
Comment (bumi) [inline src/app/components/AllowanceMenu/index.tsx:33]: maybe we also call this lnurlAuth? like the attribute that we set?

File: src/app/components/AllowanceMenu/index.tsx (hunk 1 above)
Comment (bumi) [reply src/app/components/AllowanceMenu/index.tsx:152]: +1

**Review by escapedcat — APPROVED** [review]

File: src/i18n/locales/en/translation.json [hunk 3]
```diff
@@ -570,8 +570,12 @@
       "new_budget": {
         "label": "New budget"
       },
-      "set_budget": {
-        "title": "Set a new budget",
+      "enable_login": {
+        "title": "Enable website login",
+        "subtitle": "Authenticate using <0>lnurlAuth</0>"
```
Comment (bumi) [inline src/i18n/locales/en/translation.json:575]: the text needs some update. 

```suggestion
        "subtitle": "Automatically login without confirmation when the website requests."
```

**Review by bumi — APPROVED** [review]

File: src/i18n/locales/en/translation.json (hunk 3 above)
Comment (im-adithya) [reply src/i18n/locales/en/translation.json]: Done!

File: src/app/components/AllowanceMenu/index.tsx [hunk 4]
```diff
@@ -138,14 +143,35 @@ function AllowanceMenu({ allowance, onEdit, onDelete }: Props) {
                 onChange={(e) => setBudget(e.target.value)}
               />
             </div>
+            <Setting
+              title={t("enable_login.title")}
+              subtitle={
+                <Trans
```
Comment (escapedcat) [inline src/app/components/AllowanceMenu/index.tsx:149]: Now that the string was changed do we still need the `Trans` here?

Comment (im-adithya) [reply src/app/components/AllowanceMenu/index.tsx:149]: Oops! Changing it

Comment (im-adithya) [reply src/app/components/AllowanceMenu/index.tsx]: Done!

**Comment by MoritzKa** [comment]
a bit late to the party. But it works for me on LNMarkets and bolt.fun.

**Pull request merged**